src/main/java/org/springframework/data/neo4j/core/ReactiveNeo4jTemplate.java

```diff
@@ -256,7 +256,7 @@ private Object convertIdValues(@Nullable Neo4jPersistentProperty idProperty, Obj
 						if (entityMetaData.isUsingInternalIds()) {
 							propertyAccessor.setProperty(entityMetaData.getRequiredIdProperty(), internalId);
 						}
-					}).then(Mono.defer(() -> processRelations(entityMetaData, instance, propertyAccessor, inDatabase, isNewEntity)));
+					}).flatMap(internalId -> processRelations(entityMetaData, instance, internalId, propertyAccessor, inDatabase, isNewEntity));
```

Comment:
That changes semantics… It will process relations now in parallel to the original publisher.

Reply:
Should this then be wrapped into a `flapMap(internalId -> Mono.defer()....`?
I thought the same but found the other option not really appealing and things are still working as expected.